### DISCUSSION TITLE
How to install a specific revision of a package.

### DISCUSSION THREAD

**Discussion by madelaney** [mason-org/mason.nvim#880] [Q&A] [answered]
All,

I have `mason` and `mason-lspconfig`, setup and installing some packages. However, in one of the packages there's a newly introduced bug. I'm trying to work around this bug by downgrading to a previous tag (i.e the latest is `1.3.0` but I want to install `1.2.0`).

I tried to look over the docs for how to install a specific revision but either I was using the wrong term, or I'm blind as I had no luck finding it.

- Mike D.

**Comment by williamboman** [comment, marked as answer]
Hello! You can specify a version when installing a package, like so:

- via the command interface: `:MasonInstall package@v1.0.0`
- via the Lua API: `pkg:install { version = "v1.0.0" }`

mason-lspconfig also supports specifying versions in the `ensure_installed` table: 
```lua
require("mason-lspconfig").setup {
  ensure_installed = { "package@v1.0.0" }
}
```

**Reply by madelaney** [reply]
Awesome. Thank you sir!

**Reply by Drew-Daniels** [reply]
In case this trips up anyone else, I had to remove the `v` from the version to get my package to install correctly this way:
`:MasonInstall package@1.0.0`

**Reply by srgsanky** [reply]
+1 had to remove the v in version.

The equivalent when using ensure_installed is 

```lua
require('mason-lspconfig').setup {
  ensure_installed = { 'lua_ls@3.7.4' },
}
```

Here is sample lua to do the same using APIs

```lua
local mason_registry = require 'mason-registry'
local package = mason_registry.get_package 'lua-language-server'
local lua_ls_version = '3.7.4'
if not package:is_installed() then
  package:install { version = lua_ls_version }
end
```
